attestation/factory.go

```diff
@@ -86,7 +86,20 @@ func FactoryByName(name string) (registry.FactoryFunc[Attestor], bool) {
 	return registrationEntry.Factory, ok
 }
 
+func AddAttestor(nameOrType string) (Attestor, error) {
```

Comment:
One quick thing -- The Attestors function currently would be more equivalent to a GetAttestors function -- it's returning a slice of instantiated Attestors from the factory.

AddAttestor, to me, seems to imply we're adding more attestors to the factory. Maybe `GetAttestor` or just `Attestor` is an appropriate name? I tend to avoid `GetXXXX` named functions but that's just my 2 cents.

Reply:
Sorry if I am getting this wrong, but I was under the impression that `AddAttestor(s)` made sense here because we are providing a `string` which defines the attestor we wish to instantiate, it instantiates it and *then* returns it. `GetAttestor` suggests (to me at least) that we are fetching something that has already been instantiated, hence why I went with `AddAttestor`.

Reply:
oh scrap that, I see what you mean.

Reply:
I think that while I am impartial with whether at this stage it should be `Attestors` or `AddAttestors`, that isn't enough to go and change the function name. I will revert it back to the original naming.

Reply:
I'm not married to the naming, for what it's worth. Just giving ya my thought process :)

Reply:
Ah I see. If the new function is `func Attestor` that collides with the interface. Let me know if you're happy with me keeping it as `AddAttestors` or if you wish for it to be named `GetAttestors`

Reply:
I think `GetXXXX` makes more sense to me in this context, but I am curious about what others things @jkjell

Reply:
Or, since we use `New` in other contexts to indicate we're creating a new instance of an object, we could use `NewAttestor` and `NewAttestors` .

Reply:
I am happy to switch this to `GetAttestor(s)` 😃